packages/tools/devtools/devtools-view/src/TelemetryUtils.ts

```diff
@@ -48,3 +48,46 @@ export class ConsoleVerboseLogger implements ITelemetryBaseLogger {
 		this.baseLogger?.send(event);
 	}
 }
+
+/**
+ * Key for the local storage entry that stores the usage telemetry opt-in setting.
+ */
+const telemetryOptInKey: string = "fluid:devtools:telemetry:optIn";
+
+/**
+ * Hook for getting and setting the usage telemetry opt-in setting, backed by brower's local storage.
+ * @returns A tuple (React state) with the current value and a setter for the value.
+ */
+export const useTelemetryOptIn = (): [boolean, React.Dispatch<React.SetStateAction<boolean>>] => {
+	const [value, setValue] = React.useState(() => {
+		return getStorageValue(telemetryOptInKey, true);
+	});
+
+	React.useEffect(() => {
+		localStorage.setItem(telemetryOptInKey, value.toString());
+	}, [value]);
```

Comment:
This effect will only re-run when `value` is changed via `setValue`, but won't update if the local storage value is changed otherwise. Not sure this case matters that much, but maybe there's a window hook we can subscribe to for this? Otherwise, might be worth noting the limitation in the docs.